addons/sale_pdf_quote_builder/models/sale_order.py

```diff
@@ -62,10 +62,10 @@ def get_update_included_pdf_params(self):
         ) or {}
 
         headers_available = self.available_product_document_ids.filtered(
-            lambda doc: doc.document_type == 'header'
+            lambda doc: doc.document_type == 'header' and doc.check_access('read')
```

Comment:
You probably want to use `has_access` here to avoid raising if the user doesn't have access to the given document.